tools/cp-utility/src/main.rs

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+use std::collections::VecDeque;
+use std::env;
+use std::fs;
+use std::io;
+use std::os::unix;
+use std::path::Path;
+use std::path::PathBuf;
+use std::process;
+
+/// A type of copy operation
+#[derive(Debug, PartialEq)]
+enum CopyType {
+    /// equivalent to cp <source> <dest>
+    SingleFile,
+    /// equivalent to cp -a <source> <dest>
+    Archive,
+}
+
+/// Encapsulate a copy operation
+struct CopyOperation {
+    /// The source path
+    source: PathBuf,
+    /// The destination path
+    destination: PathBuf,
+    /// The type of copy being performed
+    copy_type: CopyType,
+}
+
+/// Parse command line arguments and transform into `CopyOperation`
+fn parse_args(args: Vec<&str>) -> io::Result<CopyOperation> {
+    if !(args.len() == 3 || args.len() == 4 && args[1].eq("-a")) {
+        return Err(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "Invalid parameters. Expected cp [-a] <source> <destination>",
+        ));
+    }
+
+    if args.len() == 4 {
+        return Ok(CopyOperation {
+            source: PathBuf::from(args[2]),
+            destination: PathBuf::from(args[3]),
+            copy_type: CopyType::Archive,
+        });
+    }
+
+    Ok(CopyOperation {
+        source: PathBuf::from(args[1]),
+        destination: PathBuf::from(args[2]),
+        copy_type: CopyType::SingleFile,
+    })
+}
+
+/// Execute the copy operation
+fn do_copy(operation: CopyOperation) -> io::Result<()> {
+    match operation.copy_type {
+        CopyType::Archive => copy_archive(&operation.source, &operation.destination)?,
+        CopyType::SingleFile => fs::copy(&operation.source, &operation.destination).map(|_| ())?,
+    };
+    Ok(())
+}
+
+// Execute the recursive type of copy operation
+fn copy_archive(source: &Path, dest: &Path) -> io::Result<()> {
+    let mut stack = VecDeque::new();
```

Comment:
https://github.com/aws-observability/aws-otel-java-instrumentation/blob/main/tools/cp-utility/src/main.rs#L79-L87

I still don't get how we can have a pre-existing `/otel-auto-instrumentation-python` folder that would cause the code change to fix the problem. The target container shouldn't have this folder to begin with and should only have this when the `adot-autoinstrumentation-python` image is mounted via the `init` container either by the CW Operator or manually.

Reply:
Also, if this is the bug fix, we should consider porting the fix back to adot java.
https://github.com/aws-observability/aws-otel-java-instrumentation/blob/main/tools/cp-utility/src/main.rs#L79-L87

Can you cut a backlog item for this task?

Reply:
Yeah I'm not sure why the folder is there either. Running the sample app pod without the operator doesn't have the `/otel-auto-instrumentation-python` folder. Do you think this warrants further investigation?

Reply:
Confirmed that this is expected behavior with Mengyi